File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/ev3lejosv1/sensors_ht.java
package generated.main;

import de.fhg.iais.roberta.runtime.*;
import de.fhg.iais.roberta.runtime.ev3.*;

import de.fhg.iais.roberta.mode.general.*;
import de.fhg.iais.roberta.mode.action.*;
import de.fhg.iais.roberta.mode.sensor.*;
import de.fhg.iais.roberta.mode.action.ev3.*;
import de.fhg.iais.roberta.mode.sensor.ev3.*;

import de.fhg.iais.roberta.components.*;

import java.util.LinkedHashSet;
import java.util.HashMap;
import java.util.Set;
import java.util.Map;
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

import lejos.remote.nxt.NXTConnection;

public class NEPOprog {
    private static Configuration brickConfiguration;

    private Set<UsedSensor> usedSensors = new LinkedHashSet<UsedSensor>(Arrays.asList(new UsedSensor(SensorPort.S2, SensorType.COMPASS, CompassSensorMode.CALIBRATE), new UsedSensor(SensorPort.S2, SensorType.COMPASS, CompassSensorMode.ANGLE), new UsedSensor(SensorPort.S2, SensorType.COMPASS, CompassSensorMode.COMPASS), new UsedSensor(SensorPort.S4, SensorType.IRSEEKER, IRSeekerSensorMode.MODULATED), new UsedSensor(SensorPort.S4, SensorType.IRSEEKER, IRSeekerSensorMode.UNMODULATED), new UsedSensor(SensorPort.S3, SensorType.HT_COLOR, HiTecColorSensorV2Mode.COLOUR), new UsedSensor(SensorPort.S3, SensorType.HT_COLOR, HiTecColorSensorV2Mode.LIGHT), new UsedSensor(SensorPort.S3, SensorType.HT_COLOR, HiTecColorSensorV2Mode.AMBIENTLIGHT), new UsedSensor(SensorPort.S3, SensorType.HT_COLOR, HiTecColorSensorV2Mode.RGB)));
    private Hal hal = new Hal(brickConfiguration, usedSensors);

    private void ____sensors() {
        hal.hiTecCompassStartCalibration(SensorPort.S2);
        hal.waitFor(40000);
        hal.hiTecCompassStopCalibration(SensorPort.S2);
        hal.drawText(String.valueOf(hal.getHiTecCompassAngle(SensorPort.S2)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getHiTecCompassCompass(SensorPort.S2)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getHiTecIRSeekerModulated(SensorPort.S4)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getHiTecIRSeekerUnmodulated(SensorPort.S4)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getHiTecColorSensorV2Colour(SensorPort.S3)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getHiTecColorSensorV2Light(SensorPort.S3)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getHiTecColorSensorV2Ambient(SensorPort.S3)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getHiTecColorSensorV2Rgb(SensorPort.S3)), ___numberVar, ___numberVar);
    }

    private void ____waitUntil() {
        while ( true ) {
            if ( hal.getHiTecCompassAngle(SensorPort.S2) < 30 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getHiTecCompassCompass(SensorPort.S2) < 30 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getHiTecIRSeekerModulated(SensorPort.S4) < 30 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getHiTecIRSeekerUnmodulated(SensorPort.S4) < 30 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getHiTecColorSensorV2Colour(SensorPort.S3) == PickColor.RED ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getHiTecColorSensorV2Light(SensorPort.S3) < 50 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getHiTecColorSensorV2Ambient(SensorPort.S3) < 50 ) {
                break;
            }
            hal.waitFor(15);
        }
    }

    public static void main(String[] args) {
        try {
             brickConfiguration = new EV3Configuration.Builder()
                .setWheelDiameter(5.6)
                .setTrackWidth(18.0)
                .addSensor(SensorPort.S2, new Sensor(SensorType.COMPASS))
                .addSensor(SensorPort.S3, new Sensor(SensorType.HT_COLOR))
                .addSensor(SensorPort.S4, new Sensor(SensorType.IRSEEKER))
                .build();

            new NEPOprog().run();
        } catch ( Exception e ) {
            Hal.displayExceptionWaitForKeyPress(e);
        }
    }

    float ___numberVar = 0;
    boolean ___booleanVar = true;
    String ___stringVar = "";
    PickColor ___colourVar = PickColor.WHITE;
    NXTConnection ___connectionVar = null;
    ArrayList<Float> ___numberList = new ArrayList<>(Arrays.asList((float) 0, (float) 0));
    ArrayList<Boolean> ___booleanList = new ArrayList<>(Arrays.<Boolean>asList(true, true));
    ArrayList<String> ___stringList = new ArrayList<>(Arrays.<String>asList("", ""));
    ArrayList<PickColor> ___colourList = new ArrayList<>(Arrays.<PickColor>asList(PickColor.WHITE, PickColor.WHITE));
    ArrayList<NXTConnection> ___connectionList = new ArrayList<>(Arrays.<NXTConnection>asList(___connectionVar, ___connectionVar));

    public void run() throws Exception {
        hal.startLogging();
        ____sensors();
        hal.closeResources();

    }
}

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/ev3dev/sensors_ht.py
#!/usr/bin/python

from __future__ import absolute_import
from roberta.ev3 import Hal
from ev3dev import ev3 as ev3dev
import math
import os
import time

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

_brickConfiguration = {
    'wheel-diameter': 5.6,
    'track-width': 18.0,
    'actors': {
    },
    'sensors': {
        '2':Hal.makeCompassSensor(ev3dev.INPUT_2),
        '3':Hal.makeHTColorSensorV2(ev3dev.INPUT_3),
        '4':Hal.makeIRSeekerSensor(ev3dev.INPUT_4),
    },
}
hal = Hal(_brickConfiguration)

___numberVar = 0
___booleanVar = True
___stringVar = ""
___colourVar = 'white'
___connectionVar = None
___numberList = [0, 0]
___booleanList = [True, True]
___stringList = ["", ""]
___colourList = ['white', 'white']
___connectionList = [___connectionVar, ___connectionVar]
def ____sensors():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList

    hal.drawText(str(hal.getHiTecCompassSensorValue('2', 'angle')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getHiTecCompassSensorValue('2', 'compass')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getHiTecIRSeekerSensorValue('4', 'AC')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getHiTecIRSeekerSensorValue('4', 'DC')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getHiTecColorSensorV2Colour('3')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getHiTecColorSensorV2Light('3')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getHiTecColorSensorV2Ambient('3')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getHiTecColorSensorV2Rgb('3')), ___numberVar, ___numberVar)

def ____waitUntil():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    while True:
        if hal.getHiTecCompassSensorValue('2', 'angle') < 30:
            break
        hal.waitFor(15)
    while True:
        if hal.getHiTecCompassSensorValue('2', 'compass') < 30:
            break
        hal.waitFor(15)
    while True:
        if hal.getHiTecIRSeekerSensorValue('4', 'AC') < 30:
            break
        hal.waitFor(15)
    while True:
        if hal.getHiTecIRSeekerSensorValue('4', 'DC') < 30:
            break
        hal.waitFor(15)
    while True:
        if hal.getHiTecColorSensorV2Colour('3') == 'red':
            break
        hal.waitFor(15)
    while True:
        if hal.getHiTecColorSensorV2Light('3') < 50:
            break
        hal.waitFor(15)
    while True:
        if hal.getHiTecColorSensorV2Ambient('3') < 50:
            break
        hal.waitFor(15)

def run():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    ____sensors()

def main():
    try:
        run()
    except Exception as e:
        hal.drawText('Fehler im EV3', 0, 0)
        hal.drawText(e.__class__.__name__, 0, 1)
        hal.drawText(str(e), 0, 2)
        hal.drawText('Press any key', 0, 4)
        while not hal.isKeyPressed('any'): hal.waitFor(500)
        raise

if __name__ == "__main__":
    main()